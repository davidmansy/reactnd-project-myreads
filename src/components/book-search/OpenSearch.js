import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

function OpenSearch(props) {
  return (
    <div className="open-search">
      <Link to={props.searchPath}>Add a book</Link>
    </div>
  );
}

OpenSearch.propTypes = {
  searchPath: PropTypes.string.isRequired
};

export default OpenSearch;
