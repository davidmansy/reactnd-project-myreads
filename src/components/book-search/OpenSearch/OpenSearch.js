import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

function OpenSearch(props) {
  const { searchPath, addControlLabel } = props;

  return (
    <div className="open-search">
      <Link to={searchPath}>{addControlLabel}</Link>
    </div>
  );
}

OpenSearch.propTypes = {
  searchPath: PropTypes.string.isRequired,
  addControlLabel: PropTypes.string.isRequired
};

export default OpenSearch;
