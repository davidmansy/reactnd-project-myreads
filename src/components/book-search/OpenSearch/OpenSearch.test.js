import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import OpenSearch from './OpenSearch';
import { Link, MemoryRouter } from 'react-router-dom';

describe('OpenSearch', () => {
  const props = { searchPath: '/path', addControlLabel: 'add' };

  it('renders', () => {
    const container = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <OpenSearch {...props} />
      </MemoryRouter>,
      container
    );
    ReactDOM.unmountComponentAtNode(container);
  });
  it('displays the correct Link', () => {
    const wrapper = shallow(
      <OpenSearch searchPath="/path" addControlLabel="add" />
    );
    const link = <Link to="/path">add</Link>;
    expect(wrapper.contains(link)).toEqual(true);
  });
  it('snapshots', () => {
    const testRenderer = TestRenderer.create(
      <MemoryRouter>
        <OpenSearch {...props} />
      </MemoryRouter>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
    testRenderer.unmount();
  });
});
