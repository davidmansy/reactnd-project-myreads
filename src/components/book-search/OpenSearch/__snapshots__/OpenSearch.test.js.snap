// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenSearch snapshots 1`] = `
<div
  className="open-search"
>
  <a
    href="/path"
    onClick={[Function]}
  >
    add
  </a>
</div>
`;
